data/ignition/files/usr/local/bin/start-agent.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+source common.sh
+
+echo "Waiting for infra-env-id to be available"
+INFRA_ENV_ID=""
+until [[ $INFRA_ENV_ID != "" && $INFRA_ENV_ID != "null" ]]; do
+    sleep 5
+    printf 'Querying assisted-service for infra-env-id...'
+    INFRA_ENV_ID=$(curl {{.ServiceBaseURL}}/api/assisted-install/v2/infra-envs | jq .[0].id)
```

Comment:
```suggestion
    INFRA_ENV_ID=$(curl '{{.ServiceBaseURL}}/api/assisted-install/v2/infra-envs' | jq .[0].id)
```

IIUC, {{.ServiceBaseURL}} is to be rendered by golang. Single quoting makes the shellcheck linter happy and look better as well